apps/server/src/modules/files-storage/repo/filerecord.repo.ts

```diff
@@ -46,11 +51,50 @@ export class FileRecordRepo extends BaseRepo<FileRecord> {
 		return result;
 	}
 
-	async findBySecurityCheckRequestToken(token: string): Promise<FileRecord> {
+	public async findBySecurityCheckRequestToken(token: string): Promise<FileRecord> {
 		// Must also find expires in future. Please do not add .byExpires().
 		const scope = new FileRecordScope().bySecurityCheckRequestToken(token);
+		const fileRecord = await this.findOne(scope);
 
-		const fileRecord = await this.findOneOrFail(scope);
+		return fileRecord;
+	}
+
+	public async delete(fileRecords: FileRecord[]): Promise<void> {
+		const entities = await this.find(fileRecords);
```

Comment:
I could also think of another approach. Why do we need the find? FileRecordPropertiesFactory.buildFromDtos is already a mapper from dto to fileRecord. Why not having a version that also maps the id? And create the entites that way without a find?